File: haushaltsbuch/src/test/java/org/github/haschi/infrastruktur/Abfragekonfiguration.java
package org.github.haschi.infrastruktur;

import org.axonframework.commandhandling.gateway.CommandGateway;
import org.axonframework.config.Configuration;
import org.axonframework.config.DefaultConfigurer;
import org.github.haschi.haushaltsbuch.projektion.InventarProjektion;
import org.picocontainer.Startable;

public class Abfragekonfiguration implements Startable
{

    public CommandGateway commandGateway()
    {
        return konfiguration.commandGateway();
    }

    private final Configuration konfiguration;

    public Abfragekonfiguration(final EventStoreLieferant storagelieferant)
    {
        this.konfiguration = DefaultConfigurer.defaultConfiguration()
                .registerCommandHandler(InventarProjektion::new)
                .configureEventStore(storagelieferant::eventBus)
                .buildConfiguration();
    }

    @Override
    public void start()
    {
        konfiguration.start();
    }

    @Override
    public void stop()
    {
        konfiguration.shutdown();
    }
}

File: haushaltsbuch/src/test/kotlin/org/github/haschi/infrastruktur/Abfragekonfiguration.kt
package org.github.haschi.infrastruktur

import org.axonframework.commandhandling.gateway.CommandGateway
import org.axonframework.config.Configuration
import org.axonframework.config.DefaultConfigurer
import org.github.haschi.haushaltsbuch.projektion.InventarProjektion
import org.picocontainer.Startable

class Abfragekonfiguration(storagelieferant: EventStoreLieferant) : Startable
{

    fun commandGateway(): CommandGateway
    {
        return konfiguration.commandGateway()
    }

    private val konfiguration: Configuration = DefaultConfigurer.defaultConfiguration()
            .registerCommandHandler({ InventarProjektion(EreignisLieferant(it.eventStore())) })
            .configureEventStore({ storagelieferant.eventBus(it) })
            .buildConfiguration()

    override fun start()
    {
        konfiguration.start()
    }

    override fun stop()
    {
        konfiguration.shutdown()
    }
}
